packages/keyring-controller/src/KeyringController.ts

```diff
@@ -585,16 +813,35 @@ export class KeyringController extends BaseController<
    */
   async exportAccount(password: string, address: string): Promise<string> {
     await this.verifyPassword(password);
-    return this.#keyring.exportAccount(address);
+
+    const keyring = (await this.getKeyringForAccount(
+      address,
+    )) as EthKeyring<Json>;
+    if (!keyring.exportAccount) {
+      throw new Error(KeyringControllerError.UnsupportedExportAccount);
+    }
+
+    return await keyring.exportAccount(normalize(address) as Hex);
   }
 
   /**
    * Returns the public addresses of all accounts for the current keyring.
    *
    * @returns A promise resolving to an array of addresses.
    */
-  getAccounts(): Promise<string[]> {
-    return this.#keyring.getAccounts();
+  async getAccounts(): Promise<string[]> {
+    const keyrings = this.#keyrings;
+
+    const keyringArrays = await Promise.all(
+      keyrings.map(async (keyring) => keyring.getAccounts()),
+    );
+    const addresses = keyringArrays.reduce((res, arr) => {
+      return res.concat(arr);
+    }, []);
+
+    // Cast to `Hex[]` here is safe here because `addresses` has no nullish
```

Comment:
Just a thought for later: it would be nice if we could add an overload to `normalize` so it returns Hex if given something non-nullish, so we don't have to cast here.